src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,105 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int DEFAULT_SIZE = 0;
+    private int quantity = DEFAULT_SIZE;
+    private Object[] elements = new Object[DEFAULT_CAPACITY];
+
     @Override
     public void add(T value) {
-
+        growIfArrayFull();
+        elements[quantity] = value;
+        quantity += 1;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkInRangeForAddition(index);
+        growIfArrayFull();
+        System.arraycopy(elements, index, elements, index + 1, quantity - index);
+        elements[index] = value;
+        quantity += 1;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkInRangeForGet(index);
+        return (T) (elements[index]);
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkInRangeForGet(index);
+        elements[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkInRangeForGet(index);
+        Object temp = elements[index];
+
+        if (index != quantity - 1) {
+            System.arraycopy(elements, index + 1, elements, index, quantity - index + 1);
```

Comment:
The 'System.arraycopy()' method copies 'quantity - index + 1' elements from the 'elements' array. This can lead to an 'ArrayIndexOutOfBoundsException' because it tries to access the array element at 'quantity + 1' index which is out of bounds. It should copy 'quantity - index' elements instead.